ui/src/ducks/app/monitoring.js

```diff
@@ -39,23 +42,40 @@ export const setClusterStatusAction = payload => {
   return { type: SET_CLUSTER_STATUS, payload };
 };
 
-export function* fetchAlerts() {
-  const api = yield select(state => state.config.api);
-  const result = yield call(getAlerts, api.url_prometheus);
-  if (!result.error) {
-    yield put(setAlertsAction(result.data.data.alerts));
-  }
-}
-
 export function* fetchClusterStatus() {
-  const api = yield select(state => state.config.api);
-  const result = yield call(getClusterStatus, api.url_prometheus);
-  if (!result.error) {
-    yield put(setClusterStatusAction(result.data.data.result));
+  const clusterHealth = {
+    status: '',
+    label: ''
+  };
+  const resultAlerts = yield call(getAlerts); // Check if Prometheus API is available
+
+  if (!resultAlerts.error) {
+    yield put(setAlertsAction(resultAlerts.data.alerts));
+
+    const result = yield call(getClusterStatus);
+
+    if (!result.error) {
+      if (result.data.result && result.data.result.length) {
+        clusterHealth.status = CLUSTER_STATUS_UP;
+        clusterHealth.statusLabel = intl.translate('cluster_up_and_running');
+      } else {
+        clusterHealth.status = CLUSTER_STATUS_DOWN;
+        clusterHealth.statusLabel = intl.translate('down');
+      }
+    }
+  } else {
+    clusterHealth.status = CLUSTER_STATUS_ERROR;
+    if (resultAlerts.error.response) {
+      clusterHealth.statusLabel = `Prometheus - ${
```

Comment:
Since that one is not translated (and probably cannot be), I think it should be handled differently (not use the same state key).